modin/engines/base/frame/axis_partition.py

```diff
@@ -226,25 +242,44 @@ def deploy_axis_func(
 
     @classmethod
     def deploy_func_between_two_axis_partitions(
-        cls, axis, func, num_splits, len_of_left, kwargs, *partitions
+        cls, axis, func, num_splits, len_of_left, other_shape, kwargs, *partitions
     ):
         """Deploy a function along a full axis between two data sets in Ray.
 
-        Args:
+        Parameters
+        ----------
             axis: The axis to perform the function along.
             func: The function to perform.
             num_splits: The number of splits to return
                 (see `split_result_of_axis_func_pandas`).
             len_of_left: The number of values in `partitions` that belong to the
                 left data set.
+            other_shape: The shape of right frame in terms of partitions
             kwargs: A dictionary of keyword arguments.
             partitions: All partitions that make up the full axis (row or column)
                 for both data sets.
 
-        Returns:
+        Returns
+        -------
             A list of Pandas DataFrames.
         """
         lt_frame = pandas.concat(partitions[:len_of_left], axis=axis, copy=False)
-        rt_frame = pandas.concat(partitions[len_of_left:], axis=axis, copy=False)
+
+        rt_parts = partitions[len_of_left:]
+
+        # reshaping flattened `rt_parts` array into with shape `other_shape`
+        combined_axes = [
```

Comment:
Can you describe a little bit this logic and what is this needed for?

Reply:
I've rewrited this part a bit, this is how it's working now:

Here is the order of how partitions of broadcasted frame will be located in `rt_parts` for each axis:
![image](https://user-images.githubusercontent.com/62142979/94059390-01596880-fdeb-11ea-9548-52b85a101ada.png)
as we can see:
```python
other[axis] == len(list_of_blocks)
other[axis ^ 1] == len(other_axis_partitions)  # amount of axis partitions
```
So here we're just building our broadcasted frame from a flattened array.
`combined_axis` will contain an array of serialized axis partitions, and `rt_frame` will contain the whole serialized broadcasted frame